### PR TITLE
Task/SS-6231 sdTF SDK: new type-hints geometry.transformlist and JSON

### PR DESCRIPTION
**Pull request by matthiasreisacher**
[Jira SS-6231](https://shapediver.atlassian.net/browse/SS-6231)
[Jira SS-5995](https://shapediver.atlassian.net/browse/SS-5995)

This PR contains the following commits:
* Some minor refactoring and a small bug fix in some unit tests.
* SS-6231: sdTF SDK: new type-hints `geometry.transformlist` and `JSON`
* SS-5995: sdTF SDK: extend typeHints and types for `geometry.point` and `geometry.vector`

### CODE REVIEW

**Comment by matthiasreisacher** [comment]
Open Todo: Extend new types in sdTF specification.
I will do this after this PR has been merged.

File: libs/sdk.sdtf-geometry/__tests__/integration_tests/type_vector4d.test.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import { ISdtfReadableContentComponent, SdtfGeometryTypeHintName } from "@shapediver/sdk.sdtf-core"
+import { create, SdtfSdk } from "../../../../packages/sdk.sdtf-v1"
+import { SdtfGeometryTypeGuard, SdtfGeometryTypeIntegration, SdtfGeometryVector4d } from "../../src"
+
+describe("type vector3d", function () {
```
Comment (snabela) [inline libs/sdk.sdtf-geometry/__tests__/integration_tests/type_vector4d.test.ts:5]: @matthiasreisacher  a minor copy-paste error, should be vector4d I think

File: libs/sdk.sdtf-geometry/__tests__/integration_tests/type_vector4d.test.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        sdk = await create({ integrations: [ new SdtfGeometryTypeIntegration() ] })
+    })
+
+    // TODO remove skip when the test_data file has been created via Grasshopper
+    test.skip("read and get content; should not throw", async () => {
+        const asset = await sdk.createParser().readFromFile("./test_data/geometry_vector4d.sdtf")
+        const data = await asset.items[0].getContent()
+        expect(data).toStrictEqual(content)
+        SdtfGeometryTypeGuard.assertVector(data)
+        SdtfGeometryTypeGuard.assertVector3d(data)
```
Comment (snabela) [inline libs/sdk.sdtf-geometry/__tests__/integration_tests/type_vector4d.test.ts:20]: @matthiasreisacher should be 4d here

File: libs/sdk.sdtf-geometry/src/SdtfGeometryTypeGuard.ts [hunk 3]
```diff
@@ -321,7 +322,7 @@ export class SdtfGeometryTypeGuard {
 
     /** Returns `true` when the given value is of type `SdtfGeometryTypeHintName.GEOMETRY_VECTOR` with 2 or 3 elements. */
```
Comment (snabela) [inline libs/sdk.sdtf-geometry/src/SdtfGeometryTypeGuard.ts:323]: @matthiasreisacher 2, 3, or 4 elements

**Review by snabela — APPROVED** [review]
Thank you @matthiasreisacher !

**Pull request merged**